src/main/java/com/redhat/foreman/ForemanSharedNodeCloud.java

```diff
@@ -152,9 +152,8 @@ public boolean canProvision(Label label) {
         Map<String, String> hostsMap = getForemanAPI().getCompatibleHosts();
         Set<String> hosts = hostsMap.keySet();
         for (String host: hosts) {
-            boolean match = label.matches(Label.parse(hostsMap.get(host)));
-            if (match) {
-                    return true;
+            if (label == null || label.matches(Label.parse(hostsMap.get(host)))) {
```

Comment:
1. If label is null, should we *not* want to provision?
2. We should maybe just chk for null at start of method so we do not call the API for nothing